### PR TITLE
=str #19818 Flow.transform should be deprected too

### PR DESCRIPTION
**Pull request by agolubev**
Ref #19818

### CODE REVIEW

**Comment by johanandren** [comment]
OK TO TEST

**Comment by johanandren** [comment]
LGTM

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 1]
```diff
@@ -991,6 +991,7 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    * This operator makes it possible to extend the `Flow` API when there is no specialized
    * operator that performs the transformation.
    */
+  @deprecated("As Stage is deprecated.", "2.4.2")
```
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Will be as of 2.4.3 though. We can't really deprecate it after the fact?

**Comment by hepin1989** [comment]
should we add a new method
`def transform[T](mkStage: () ⇒ GraphStage[FlowShape[Out,T]]): Repr[T]`

**Comment by johanandren** [comment]
No, `GraphStages` are meant to be re-usable blueprint constructs, we should not endorse writing them in only-usable-once ways.

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala (hunk 1 above)
Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: And, I think the message should guide the user about what to do instead.

Comment (agolubev) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: And, I think the message should guide the user about what to do instead.>

Is it FlowOps.via?

Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Maybe "Use via(GraphStage) instead"?

Comment (agolubev) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Ok

**Comment by johanandren** [comment]
LGTM!

**Comment by drewhk** [comment]
LGTM

**Pull request merged**